lib/types.nix

```diff
@@ -397,7 +397,7 @@ rec {
 
     listOf = elemType: mkOptionType rec {
       name = "listOf";
-      description = "list of ${elemType.description}";
+      description = "list of (${elemType.description})";
```

Comment:
Atomic types don't need parentheses in container-like types.

Reply:
See comment, improving this logic would be nice but needs the types themselves to tell whether they want to be bracketed, I didn’t want to make that nontrivial change.

Reply:
One could do `"(list of ${elemType.description})"` and similarly for all container types. This would easily give use the correct behavior on primitive types, at the cost of having unnecessary parentheses at the top level (which could probably be stripped).

Reply:
Here's a way to do it https://github.com/NixOS/nixpkgs/pull/191353
The tests show a few examples.